apps/src/music/MusicView.jsx

```diff
@@ -0,0 +1,633 @@
+      function() {
+        this.getInput('trigger').appendField(
+          new Blockly.FieldDropdown(function() {
+            return Triggers.map(trigger => [trigger.dropdownLabel, trigger.id]);
+          }),
+          'trigger'
+        );
+      }
+    );
+
+    for (let blockType of Object.keys(MUSIC_BLOCKS)) {
+      Blockly.Blocks[blockType] = {
+        init: function() {
+          this.jsonInit(MUSIC_BLOCKS[blockType].definition);
+        }
+      };
+
+      Blockly.JavaScript[blockType] = MUSIC_BLOCKS[blockType].generator;
+    }
+
+    const container = document.getElementById('blockly-div');
+
+    this.workspace = Blockly.inject(container, {
+      // Toolbox will be programmatically generated once music manifest is loaded
+      toolbox: baseToolbox,
+      grid: {spacing: 20, length: 0, colour: '#444', snap: true},
+      theme: musicLabDarkTheme,
+      renderer: 'cdo_renderer_zelos'
+    });
+
+    this.resizeBlockly();
+
+    const xml = parseXmlElement(
+      `<xml><block type="${
+        BlockTypes.WHEN_RUN
+      }" deletable="false" x="30" y="30"></block><block type="${
+        BlockTypes.TRIGGERED_AT
+      }" deletable="false" x="500" y="30"></block></xml>`
+    );
+    Blockly.Xml.domToBlockSpace(Blockly.mainBlockSpace, xml);
+
+    Blockly.addChangeListener(Blockly.mainBlockSpace, this.onBlockSpaceChange);
+
+    this.workspace.registerButtonCallback('createVariableHandler', button => {
+      Blockly.Variables.createVariableButtonHandler(
+        button.getTargetWorkspace(),
+        null,
+        null
+      );
+    });
+  };
+
+  onBlockSpaceChange = e => {
+    // A drag event can leave the blocks in a temporarily unusable state,
+    // e.g. when a disabled variable is dragged into a slot, it can still
+    // be disabled.
+    // A subsequent non-drag event should arrive and the blocks will be
+    // usable then.
+    // It's possible that other events should similarly be ignored here.
+    if (e.type === Blockly.blockly_.Events.BLOCK_DRAG) {
+      return;
+    }
+
+    // Stop all when_run sounds that are still to play, because if they
+    // are still valid after the when_run code is re-executed, they
+    // will be scheduled again.
+    this.stopAllSoundsStillToPlay();
+
+    // Also clear all when_run sounds from the events list, because it
+    // will be recreated in its entirely when the when_run code is
+    // re-executed.
+    this.player.clearWhenRunEvents();
+
+    this.executeSong();
+
+    console.log('onBlockSpaceChange', Blockly.getWorkspaceCode());
+
+    this.analyticsReporter.onBlocksUpdated(this.workspace.getAllBlocks());
+
+    // This is a way to tell React to re-render the scene, notably
+    // the timeline.
+    this.setState({updateNumber: this.state.updateNumber + 1});
+  };
+
+  onResize = () => {
+    const windowWidth = window.innerWidth;
+    const windowHeight = window.innerHeight;
+    //const windowWidth = Math.max(window.innerWidth, window.innerHeight);
```

Comment:
this method and `resizeBlockly` have some commented out code--intentional?

Reply:
Ah, we will likely need to revive this when we do better mobile support.  I will add a comment.